src/components/sessions/activeSession/confusion/ConfusionBarometer.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Checkbox, Header } from 'semantic-ui-react'
+import { FormattedMessage } from 'react-intl'
+
+import withCSS from '../../../../lib/withCSS'
+
+const ConfusionBarometer = ({ data }) => {
+  const calculateAverage = (allData, category) => {
```

Comment:
I think this function could easily be generalized to not need a switch case at all. Generally don't do such calculations inside a component but extract the function into utils files and add tests for them. Also check whether lodash doesn't do any such thing with built in functions.